datafusion/row/src/writer.rs

```diff
@@ -326,8 +327,10 @@ pub(crate) fn write_field_date32(
     col_idx: usize,
     row_idx: usize,
 ) {
-    let from = from.as_any().downcast_ref::<Date32Array>().unwrap();
-    to.set_date32(col_idx, from.value(row_idx));
+    match as_date32_array(from) {
+        Ok(from) => to.set_date32(col_idx, from.value(row_idx)),
+        Err(e) => println!("{}", e),
```

Comment:
IMO logging here is worse than unwrapping. The function is called `write_field_date32()` and if a date32 wasn't written to something, abending is preferable to carrying on in an in-determinant state. (of course, returning an `Err` `Result` would be even more preferable).

Reply:
Yeah, I agree I prefer "unwrap()" rather than println -- `stdout` may or may not be connected to something which would be noticed

Reply:
It seems reasonable to me. I will fix it and be more careful in future PRs.